security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -560,7 +557,7 @@ func (sc *SecretManagerClient) generateNewSecret(resourceName string) (*security
 	cacheLog.Debugf("constructed host name for CSR: %s", csrHostName.String())
 	options := pkiutil.CertOptions{
 		Host:       csrHostName.String(),
-		RSAKeySize: keySize,
+		RSAKeySize: sc.configOptions.WorkloadRSAKeySize,
```

Comment:
Do we need to put some validation on the configured keysize, warn and fallback to the default if it is not valid, to prevent user shoot themselves in the foot?

Reply:
It would be good to have this, but I'm not sure if it is really needed. I don't see this being done for other envs. Also, isn't the user responsible for providing a reasonable value for this field? The GenCSR func currently checks for the minimumRSAKeySize to be 2048, similarly, maxRSAKeySize could be introduced? @jacob-delgado do you have any opinion on this?

Reply:
No real opinion. If someone is doing this they are doing this intentionally so I hope they'd not misconfigure a system. A minimum check is recommended though so they cannot create an insecure system (e.g. 512).